jwala-services/src/main/java/com/cerner/jwala/service/custom/logging/impl/JvmWinSvcAcctPasswordScrubber.java

```diff
@@ -0,0 +1,95 @@
+package com.cerner.jwala.service.custom.logging.impl;
+
+import com.cerner.jwala.common.domain.model.jvm.Jvm;
+import com.cerner.jwala.common.domain.model.ssh.DecryptPassword;
+import com.cerner.jwala.persistence.service.JvmPersistenceService;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Appender;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.apache.log4j.PatternLayout;
+import org.apache.log4j.spi.LoggingEvent;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+
+/**
+ * Scrubs JVM Windows service account passwords from the logs
+ *
+ * Created by Jedd Cuison on 6/14/2017
+ */
+@Component
+public class JvmWinSvcAcctPasswordScrubber extends PatternLayout {
+
+    public static final String REPLACEMENT = "********";
+    private final Set<String> toRemove = new HashSet<>();
+    private DecryptPassword decryptor;
+    private final String [] includedMethodsArray = {"getExecRemoteCommandReturnInfo", "runExecCommand"};
+    private final Set<String> includedMethods = new HashSet<>(Arrays.asList(includedMethodsArray));
+
+    public JvmWinSvcAcctPasswordScrubber(final JvmPersistenceService jvmPersistenceService, final DecryptPassword decryptor) {
+        this.decryptor = decryptor;
+
+        // Populate the map of items to remove in the logs
+        final List<Jvm> jvms = jvmPersistenceService.getJvms();
```

Comment:
When is this initialized? What happens to newly added JVM's with service accounts